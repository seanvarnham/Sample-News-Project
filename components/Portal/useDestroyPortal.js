import { useEffect } from "react";

const destroyRootElement = (id) => {
	const rootContainer = document.querySelector(`#${id}`);
	rootContainer.remove();
};

const useDestroyPortal = (id, keepRoot = true) => {
	useEffect(() => {
		return () => {
			if (!keepRoot) {
				destroyRootElement(id);
			}
		};
	});
	return null;
};

export default useDestroyPortal;
