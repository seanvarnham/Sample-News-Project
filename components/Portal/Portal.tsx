import { createPortal } from 'react-dom';
import usePortal from './usePortal';
import useDestroyPortal from './useDestroyPortal';

/**
 * @example
 * <Portal>
 *   <p>Portal content</p>
 * </Portal>
 */

interface IPortalProps {
  id: string;
  children: React.ReactNode;
  keepRoot?: boolean;
}

const Portal = ({ id, children, keepRoot }: IPortalProps) => {
  const domNode = usePortal(id);
  useDestroyPortal(id, keepRoot);
  if (domNode) {
    return createPortal(children, domNode);
  }
  return null;
};

export default Portal;
