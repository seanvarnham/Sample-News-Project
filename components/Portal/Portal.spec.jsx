import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import Portal from "./Portal";

describe("Portal component tests", () => {
	const renderComponent = ({ id, children }) =>
		render(<Portal id={id}>{children}</Portal>);

	it("should not render without portal element", () => {
		const { container } = render(<Portal />);
		expect(container).toMatchSnapshot();
	});
});
