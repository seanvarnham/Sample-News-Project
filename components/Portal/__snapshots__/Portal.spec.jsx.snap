// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Portal component tests should not render without portal element 1`] = `<div />`;
